### PR TITLE
Migrate 'custom taxonomies' e2e test to Playwright

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Part of #38851.
Supersedes and closes #45178.

PR migrates sidebar-permalink.test.js e2e test to Playwright.

## Why?
See #38851.

## How?
The migration is based on #45178, which I think is abandoned. I've included the author so they get credit for their contribution.

## Testing Instructions
```
npm run test:e2e:playwright -- test/e2e/specs/editor/plugins/custom-taxonomies.spec.js
```

### CODE REVIEW

File: test/e2e/specs/editor/plugins/custom-taxonomies.spec.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		const modelPanel = editorSettings.getByRole( 'button', {
+			name: 'Model',
+		} );
+
+		// Open the panel if needed.
+		if (
+			( await modelPanel.getAttribute( 'aria-expanded' ) ) === 'false'
+		) {
+			await modelPanel.click();
+		}
```
Comment (WunderBart) [inline test/e2e/specs/editor/plugins/custom-taxonomies.spec.js:38]: Is this another example of the global state bleed when running tests in parallel?

**Review by WunderBart — APPROVED** [review]
Just one non-blocking question. 🚀

File: test/e2e/specs/editor/plugins/custom-taxonomies.spec.js (hunk 1 above)
Comment (Mamaduka) [reply test/e2e/specs/editor/plugins/custom-taxonomies.spec.js:38]: Correct.

Comment (WunderBart) [reply test/e2e/specs/editor/plugins/custom-taxonomies.spec.js:38]: Ouch!

Comment (Mamaduka) [reply test/e2e/specs/editor/plugins/custom-taxonomies.spec.js:38]: This doesn't solve the persisted state issue, but maybe we should introduce the `openDocumentSettingsPanel` helper and hide implementation details there.

**Pull request merged**